src/solver/writer/zip_writer.cpp

```diff
@@ -70,12 +78,14 @@ ZipWriter::ZipWriter(std::shared_ptr<Yuni::Job::QueueService> qs,
 
 ZipWriter::~ZipWriter()
 {
-    mz_zip_writer_close(pZipHandle);
-    mz_zip_writer_delete(&pZipHandle);
+    this->finalize(false);
 }
 
 void ZipWriter::addJob(const std::string& entryPath, Yuni::Clob& entryContent)
 {
+    if (pState != ZipState::can_receive_data)
```

Comment:
We cannot make **addJob** a template method (with **entryContent**'s type as template parameter) because of polymorphism with parent class **IResultWriter**.
But what we could do is have a template private method : 
```C++
class ZipWriter : public IResultWriter
{
....
private:
    template<ContentType>
    void runJob(const std::string& entryPath, Yuni::Clob& entryContent);
}
....

template<class ContentType>
ZipWrite<ContentType>::runJob(const ContentT& entryPath, Yuni::Clob& entryContent)
{
    if (pState != ZipState::can_receive_data)
        return;

    EnsureQueueStartedIfNeeded ensureQueue(this, pQueueService);
    pQueueService->add(
      new ZipWriteJob<ContentType>(*this, entryPath, entryContent, pDurationCollector),
      Yuni::Job::priorityLow);
}
```
and call this template method inside **addJob** methods.
ok, we would have to specialize **runJob** for **IniFile** type...

Reply:
We agree on that, my comment does not contradict yours

Reply:
I've done the requested change [here](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/827/commits/ae0fa71bf6a01b0e7766d462548cbeb1d2b5f976)